linode_metadata/metadata_client.py

```diff
@@ -76,22 +68,115 @@ def __init__(
             )
 
         self.base_url = base_url
-        self.session = httpx.Client()
         self._append_user_agent = user_agent
         self.timeout = timeout
 
         self._token = token
+        self.client = None
 
         self._managed_token = managed_token
         self._managed_token_expiry_seconds = managed_token_expiry_seconds
         self._managed_token_expiry = None
 
-        self.check_connection()
+    @staticmethod
+    def _parse_response_body(response: Response, content_type: str) -> Any:
+        handlers = {
+            "application/json": response.json,
+            "text/plain": lambda: response.content,
+        }
 
-        if managed_token:
-            self.refresh_token(
-                expiry_seconds=self._managed_token_expiry_seconds,
-            )
+        if response.status_code == 204:
+            return None
+
+        handler = handlers.get(content_type.lower())
+        if handler is None:
+            raise ValueError(f"Invalid Content-Type: {content_type}")
+
+        return handler()
+
+    def _get_http_method(
+        self, method: str
+    ) -> Optional[
+        Callable[..., Optional[Union[Awaitable[Response], Response]]]
+    ]:
+        """
+        Return a callable for making the API call.
+        """
+        if not self.client:
+            raise RuntimeError("HTTP client is not defined")
+
+        method_map = {
+            "GET": self.client.get,
+            "POST": self.client.post,
+            "PUT": self.client.put,
+            "DELETE": self.client.delete,
+        }
+        return method_map.get(method.upper())
+
+    def _prepare_headers(
+        self, content_type: str, additional_headers: dict, authenticated: bool
+    ):
+        headers = {
+            "Content-Type": content_type,
+            "Accept": content_type,
```

Comment:
Is it a typo?

Reply:
It seems to be a typo in the original code base, I copied and didn't check. Thanks for pointing that out!
https://github.com/linode/py-metadata/blob/176233d3756120a0b6a6e3ee28d80d321adbb8b4/linode_metadata/metadata_client.py#L168-L171

Reply:
It's actually not a typo because they are both `application/json`. One is the content type of the request body and another is the content type it wants the response body to be. But yes, this is also problematic because `Content-Type` header typically only in requests of PUT and POST HTTP method. I updated the code to fix that.